charlib/liberty/liberty.py

```
@@ -58,29 +83,57 @@ def __repr__(self):
+        """
         try:
-            return self.groups[hash((name, identifier))]
+            return self.groups[(name, identifier)]
         except KeyError as e:
-            # Use the non-hashed key in error display
-            raise KeyError((name, identifier)) from e
+            if attributes: # Fall back to subgroup search
+                groups = self.filter_subgroups(name, identifier, **attributes)
+                groups = [g for g in groups  if g.unique_key in self.groups]
+                if len(groups) != 1:
+                    raise KeyError(f'Subgroup search found {len(groups)} matching groups!') from e
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)